include/range/v3/view/iota.hpp

```diff
@@ -44,399 +43,359 @@ namespace ranges
-            return from_ == to_;
+            return {from_, to_};
         }
         CPP_member
-        auto equal(iota_view const &that) const ->
-            CPP_ret(bool)(
-                requires Incrementable<From>)
+        auto end_cursor() const ->
+            CPP_ret(cursor)(
+                requires Same<From, To>)
         {
-            return that.from_ == from_;
+            return {to_, to_, true};
         }
         CPP_member
-        auto prev() -> CPP_ret(void)(
-            requires BidirectionalIncrementable<From>)
+        auto end_cursor() const ->
+            CPP_ret(default_sentinel_t)(
+                requires (!Same<From, To>))
         {
-            --from_;
+            return {};
         }
-        CPP_member
-        auto check_advance_(difference_type_ n) -> CPP_ret(void)(
-            requires SizedIncrementableSentinel<To, From>)
-        {
-            detail::ignore_unused(n);
-            RANGES_EXPECT(detail::iota_minus_(to_, from_) >= n);
-        }
-        template<typename = void>
-        void check_advance_(difference_type_) const
-        {
-        }
-        CPP_member
-        auto advance(difference_type_ n) -> CPP_ret(void)(
-            requires RandomAccessIncrementable<From>)
+
+        constexpr /*c++14*/ void check_bounds_(std::true_type)
         {
-            this->check_advance_(n);
-            from_ = detail::iota_plus(from_, n);
+            RANGES_EXPECT(from_ <= to_);
         }
-        CPP_member
-        auto distance_to(iota_view const &that) const ->
-            CPP_ret(difference_type_)(
-                requires RandomAccessIncrementable<From>)
+        constexpr /*c++14*/ void check_bounds_(std::false_type)
+        {}
+    public:
+        closed_iota_view() = default;
+        constexpr closed_iota_view(meta::id_t<From> from, meta::id_t<To> to)
+          : from_(std::move(from))
+          , to_(std::move(to))
         {
-            return detail::iota_minus_(that.from_, from_);
+            check_bounds_(meta::bool_<StrictTotallyOrderedWith<From, To>>{});
         }
-    public:
-        iota_view() = default;
-        constexpr iota_view(From from, To to)
-          : from_(detail::move(from)), to_(detail::move(to))
-        {}
     };
 
-    template<typename From>
-    struct iota_view<From, void>
-      : view_facade<iota_view<From, void>, infinite>
+#if RANGES_CXX_DEDUCTION_GUIDES >= RANGES_CXX_DEDUCTION_GUIDES_17
+    CPP_template(typename From, typename To)(
+        requires WeaklyIncrementable<From> && Semiregular<To> &&
+            (!Integral<From> || !Integral<To> ||
+                std::is_signed<From>::value == std::is_signed<To>::value))
+    closed_iota_view(From, To) -> closed_iota_view<From, To>;
+#endif
+
+    template<typename From, typename To /* = unreachable_sentinel_t*/>
+    struct RANGES_EMPTY_BASES iota_view
+      : view_facade<
+            iota_view<From, To>,
+            Same<To, unreachable_sentinel_t> ? infinite : finite>
```

Comment:
`finite` here should probably be `unknown`, or `is_integeral_v<From> && is_integeral_v<To> ? finite : unknown`.